### PR TITLE
refactor(match-managers): implement `custom.<customMgrName>` syntax

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Context
Improving on [this ](https://github.com/renovatebot/renovate/pull/24112) PR as I had forgotten to add `MatchManagersMigration` class to the `migration-services.ts` file,
which led to the migration not working and disabling regex dependecies :)

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — COMMENTED** [review]
is this PR currently relevant? we reverted the custom manager renaming recently because of this issue

**Comment by RahulGautamSingh** [comment]
Not sure exactly. But the last PR was reverted because of the reason I mentioned in description and this PR includes the fix for that + the migration and logic needed to handle the rename

**Comment by rarkins** [comment]
@viceice I think we reverted this PR's earlier attempt, not the full renaming?

**Comment by rarkins** [comment]
@RahulGautamSingh ready to resume?

**Comment by RahulGautamSingh** [comment]
> @RahulGautamSingh ready to resume?

Trying to fix a cyclic dependency issue :)

**Comment by RahulGautamSingh** [comment]
Test runs:
1. [match-managers-1](https://github.com/RahulGautamSingh-testing/match-manager-old/pulls) 
> -  migrates old syntax  
> - confirmed that `matchManagers` rules is working as expected, by creating group based on managers
2. [match-managers-2 ](https://github.com/RahulGautamSingh-testing/match-managers-2/issues/1)
> - confirmed that config validation errors are working thrown if invalid custom manager is passed

File: lib/config/validation-helpers/managers.ts [hunk 1]
```diff
@@ -13,7 +13,8 @@ export function check({
   if (Array.isArray(resolvedRule.matchManagers)) {
     if (
       resolvedRule.matchManagers.find(
-        (confManager) => !allManagersList.includes(confManager)
+        (confManager) =>
+          !allManagersList.includes(confManager.replace('custom.', ''))
```
Comment (secustor) [inline lib/config/validation-helpers/managers.ts:17]:
```suggestion
          !isKnownManager(confManager)
```
I think we should get a utility function for that.

**Review by secustor — COMMENTED** [review]
else LGTM

**Review by secustor — DISMISSED** [review]

**Review by secustor — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**